test/e2e/pkg/infra/docker/docker.go

```diff
@@ -0,0 +1,100 @@
+package docker
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+	"strconv"
+	"text/template"
+
+	e2e "github.com/tendermint/tendermint/test/e2e/pkg"
+	"github.com/tendermint/tendermint/test/e2e/pkg/infra"
+)
+
+var _ infra.Provider = &Provider{}
+
+// Provider implements a docker-compose backed infrastructure provider.
+type Provider struct {
+	Testnet *e2e.Testnet
+}
+
+// Setup generates the docker-compose file and write it to disk, erroring if
+// any of these operations fail.
+func (p *Provider) Setup() error {
+	compose, err := dockerComposeBytes(p.Testnet)
+	if err != nil {
+		return err
+	}
+	// nolint: gosec
+	// G306: Expect WriteFile permissions to be 0600 or less
+	err = os.WriteFile(filepath.Join(p.Testnet.Dir, "docker-compose.yml"), compose, 0644)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// dockerComposeBytes generates a Docker Compose config file for a testnet and returns the
+// file as bytes to be written out to disk.
+func dockerComposeBytes(testnet *e2e.Testnet) ([]byte, error) {
+	// Must use version 2 Docker Compose format, to support IPv6.
+	tmpl, err := template.New("docker-compose").Funcs(template.FuncMap{
+		"misbehaviorsToString": func(misbehaviors map[int64]string) string {
```

Comment:
I know you just moved this code but: I've looked everywhere and haven't found how these "misbehaviors" affect the behavior when running the tests (after digging a bit with Nenad, it looks like a leftover).
IOW, what would break if we nuke this "misbehaviors" logic?

Reply:
Looks like much of it was removed here: 07d242c46139841bf7736fd00b36705f2745ae04
Perhaps this is just remnants. Removing.